### PR TITLE
Feature mes 4294 cat d wrtc page

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES 4294
Created Cat D version of page
Modified a lot of actions, selectors and reducers to be made Cat C.
Also did Vehicle Checks modal, which is technically MES 4295 as well.

## Checklist

- [X] PR title includes the JIRA ticket number
- [X] Branch is rebased against the latest develop
- [X] Code has been tested manually
- [X] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
NOTE: Screenshots to follow.

### CODE REVIEW

File: src/pages/waiting-room-to-car/cat-d/waiting-room-to-car.cat-d.page.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+import { getUntitledCandidateName } from '../../../modules/tests/journal-data/common/candidate/candidate.selector';
+import { getTests } from '../../../modules/tests/tests.reducer';
+import { FormGroup } from '@angular/forms';
+import { QuestionProvider } from '../../../providers/question/question';
+import {
+  TellMeQuestionSelected,
+  TellMeQuestionCorrect,
+  TellMeQuestionDrivingFault,
+  QuestionOutcomes,
+} from '../../../modules/tests/test-data/cat-b/vehicle-checks/vehicle-checks.actions';
```
Comment (matthew2564) [inline src/pages/waiting-room-to-car/cat-d/waiting-room-to-car.cat-d.page.ts]: actions are from cat-b

File: src/modules/tests/test-data/cat-d/test-requirements/test-requirements.cat-d.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as testRequirementsActions from '../../cat-b/test-requirements/test-requirements.actions';
```
Comment (matthew2564) [inline src/modules/tests/test-data/cat-d/test-requirements/test-requirements.cat-d.reducer.ts]: actions from cat-b

File: src/modules/tests/test-data/cat-d/test-requirements/test-requirements.cat-d.reducer.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      return {
+        ...state,
+        [action.payload]: !state[action.payload],
+      };
+    default:
+      return state;
+  }
+}
+
+export const getTestRequirementsCatC =
```
Comment (TomBillingtonUK) [inline src/modules/tests/test-data/cat-d/test-requirements/test-requirements.cat-d.reducer.ts]: Cat C

**Review by TomBillingtonUK — APPROVED** [review]
Looks good :)

File: src/modules/tests/test-data/cat-d/test-data.cat-d.selector.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  | CatDEUniqueTypes.VehicleChecks
+  | CatD1EUniqueTypes.VehicleChecks;
+
+export const getDrivingFaultCount = (
+  data: CatDTestData, competency: Competencies) => data.drivingFaults[competency];
+
+export const getManoeuvres = (data: CatDTestData): CatDManoeuvres => data.manoeuvres;
+
+// TODO - We should pass a Manoeuvre object here instead of TestData
+export const hasManoeuvreBeenCompletedCatC = (data: CatDTestData) => {
```
Comment (mgoldson) [inline src/modules/tests/test-data/cat-d/test-data.cat-d.selector.ts]: should this be 'hasManoeuvreBeenCompletedCatD'?

File: src/modules/tests/test-data/cat-d/test-data.cat-d.selector.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      }
+    });
+  }
+
+  return correct;
+};
+
+// TODO - We should really pass a Vehicle Checks object here and not Test Data
+// TODO - Also this has to go into a provider
+export const hasVehicleChecksBeenCompletedCatC = (data: CatDTestData): boolean => {
```
Comment (mgoldson) [inline src/modules/tests/test-data/cat-d/test-data.cat-d.selector.ts:81]: should this be 'hasManoeuvreBeenCompletedCatD'?

File: src/modules/tests/test-data/cat-d/uncouple-recouple/__tests__/uncouple-recouple.cat.d1e.reducer.spec.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  ToggleUncoupleRecouple,
+  UncoupleRecoupleAddDrivingFault,
+  UncoupleRecoupleAddSeriousFault,
+  UncoupleRecoupleAddDangerousFault,
+  UncoupleRecoupleRemoveFault,
+  AddUncoupleRecoupleComment,
+} from '../../../common/uncouple-recouple/uncouple-recouple.actions';
+import { CompetencyOutcome } from '../../../../../../shared/models/competency-outcome';
+
+describe('uncoupleRecoupleCatC1EReducer' , () => {
```
Comment (mgoldson) [inline src/modules/tests/test-data/cat-d/uncouple-recouple/__tests__/uncouple-recouple.cat.d1e.reducer.spec.ts]: 'uncoupleRecoupleCatD1EReducer'

File: src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+  });
+
+  describe('showMeQuestionChanged$ effect', () => {
+    const questionNumber: number = 1;
+    const showMeQuestion: QuestionResult = {
+      code: 'S01',
+    };
+    it('should log an analyics event with show me question info', (done) => {
+
+      store$.dispatch(new testsActions.StartTest(12345, TestCategory.C));
```
Comment (mgoldson) [inline src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts]: TestCategory.D ?

File: src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+      });
+    });
+  });
+
+  describe('showMeQuestionOutComeChanged$', () => {
+    const questionOutcome: QuestionOutcome = 'P';
+    const questionNumber: number = 1;
+    it('should log an analytics event with show me question outcome info', (done) => {
+
+      store$.dispatch(new testsActions.StartTest(12345, TestCategory.C));
```
Comment (mgoldson) [inline src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts]: TestCategory.D ?

File: src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  });
+
+  describe('tellMeQuestionChanged$ effect', () => {
+    const questionNumber: number = 1;
+    const tellMeQuestion: QuestionResult = {
+      code: 'T01',
+    };
+    it('should log an analyics event with tell me question info', (done) => {
+
+      store$.dispatch(new testsActions.StartTest(12345, TestCategory.C));
```
Comment (mgoldson) [inline src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts]: TestCategory.D ?

File: src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+      });
+    });
+  });
+
+  describe('tellMeQuestionOutComeChanged$', () => {
+    const questionOutcome: QuestionOutcome = 'DF';
+    const questionNumber: number = 1;
+    it('should log an analytics event with tell me question outcome info', (done) => {
+
+      store$.dispatch(new testsActions.StartTest(12345, TestCategory.C));
```
Comment (mgoldson) [inline src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts]: TestCategory.D ?

**Review by mgoldson — APPROVED** [review]
Awesome work Ross! Just a few missed old category references to look at.

File: src/pages/waiting-room-to-car/cat-d/waiting-room-to-car.cat-d.page.ts (hunk 1 above)
Comment (rossfauldsbjss) [reply src/pages/waiting-room-to-car/cat-d/waiting-room-to-car.cat-d.page.ts]: These seem to have been copied in from the cat c version of the page. Turns out there is a method that uses these that isn't called. Removed and tested and all is fine. Removed from cat c as well

File: src/modules/tests/test-data/cat-d/test-requirements/test-requirements.cat-d.reducer.ts (hunk 2 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-data/cat-d/test-requirements/test-requirements.cat-d.reducer.ts]: Have moved these into common as all the same

File: src/modules/tests/test-data/cat-d/test-requirements/test-requirements.cat-d.reducer.ts (hunk 3 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-data/cat-d/test-requirements/test-requirements.cat-d.reducer.ts]: Done

File: src/modules/tests/test-data/cat-d/test-data.cat-d.selector.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-data/cat-d/test-data.cat-d.selector.ts]: It should. Nice catch.

File: src/modules/tests/test-data/cat-d/uncouple-recouple/__tests__/uncouple-recouple.cat.d1e.reducer.spec.ts (hunk 6 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-data/cat-d/uncouple-recouple/__tests__/uncouple-recouple.cat.d1e.reducer.spec.ts]: Done.

File: src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts (hunk 7 above)
Comment (rossfauldsbjss) [reply src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts]: Done

File: src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts (hunk 8 above)
Comment (rossfauldsbjss) [reply src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts]: done

File: src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts (hunk 9 above)
Comment (rossfauldsbjss) [reply src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts]: done

File: src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts (hunk 10 above)
Comment (rossfauldsbjss) [reply src/pages/waiting-room-to-car/cat-d/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-d.analytics.effects.spec.ts]: Done

File: src/modules/tests/test-data/cat-d/test-data.cat-d.selector.ts (hunk 5 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-data/cat-d/test-data.cat-d.selector.ts:81]: Another nice catch. It should indeed.

**Pull request merged**